editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -667,6 +667,13 @@ void VisualShaderEditor::_add_node(int p_idx, int p_op_idx) {
 		VisualShaderNode *vsn = Object::cast_to<VisualShaderNode>(ClassDB::instance(add_options[p_idx].type));
 		ERR_FAIL_COND(!vsn);
 
+		VisualShaderNodeScalarConstant *constant = Object::cast_to<VisualShaderNodeScalarConstant>(vsn);
+
+		if (constant) {
+			if (((int)add_options[p_idx].value) != -1)
```

Comment:
I think you have one pair of parentheses too much, `if ((int)add_options[p_idx].value != -1)` should be sufficient.